core/vm/evm.go

```diff
@@ -605,7 +605,7 @@ func (evm *EVM) GetVMContext() *tracing.VMContext {
 		BlockNumber: evm.Context.BlockNumber,
 		Time:        evm.Context.Time,
 		Random:      evm.Context.Random,
-		GasPrice:    evm.TxContext.GasPrice,
```

Comment:
@s1na it might be an incompatible change, but I feel like it's the right direction.

The effectiveGasPrice could be derived with BaseFee (block context) and tx gas price.

Originally, we must to invoke evm.SetTxContext before the TxStartHook, with this change, the requirement is gone

Reply:
Wait a minute. The TxStart hook: 
```
	TxStartHook = func(vm *VMContext, tx *types.Transaction, from common.Address)
```
It has the `tx`. The tx has a `GasPrice` method. Why would we have it _also_ in the `VMContext`? Gary's change makes total sense

Reply:
@maoueh Just want to mention that the GasPrice taken from the transaction object is the different with one in the `*tracing.VMContext`, the first one refers to the full gas price while the latter one refers to the effectiveGasPrice

Reply:
@rjl493456442 Yes thanks for the info. We are handling the different tx types to pick the correct gas values.

Reply:
I think we should bring it back in another tracing API version. The tracer implementation should not have to compute the effective gas price by itself. It should be handled by the tracing infrastructure.